tensorflow-frameworks/tensorflow-data/src/main/java/org/tensorflow/data/OneShotIterator.java

```diff
@@ -0,0 +1,25 @@
+package org.tensorflow.data;
+
+import org.tensorflow.Operation;
+import org.tensorflow.Output;
+import org.tensorflow.op.data.MakeIterator;
+
+import java.util.List;
+
+public class OneShotIterator {
```

Comment:
Maybe we can move this class as a `@Operator` available in `Ops`? e.g. `tf.data.oneShotIterator(makeIterator, components)`? I understand we don't have access to this op because  it uses a `func` attribute but that could replace it?

Reply:
I refactored the iteration mechanisms to capture both eager and graph iteration in `DatasetIterator`; I think the new logic is much cleaner (See the README / RFC)